### ISSUE TITLE
The `tf.linalg.svd` bug report!  Serious!

### ISSUE THREAD

**Issue by yhao-z** [tensorflow/tensorflow#55840]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.5 AND tf 2.8

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04 AND Windows 10

### Python version

3.9

### Current Behaviour?

A bug happened!
I make a example net that contains a simple `keras.Sequence` of a `Conv3D` keras layer, a `tf.linalg.svd` operation that consists of a trainable parameter `thres` and another `Conv3D` keras layer. 
When I calculate the gradients of the `Conv3D` layers, the `tf2.5` returns `nan` for the first `Conv3D` layer while the `tf2.8` seems to be normal but sometimes `tf2.8` reports error: `Segmentation Fault (core dumped)`, which I think may due to the `tf.linalg.svd`. please check it. 

run the Standalone code to reproduce the issue, and we choose a very simple input (tf.ones) and the bug can be reproduced too. 

We test it on Nvidia GV100 GPU and on `tf2.5` and `tf2.8`.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

class CONV_OP(tf.keras.layers.Layer):
    def __init__(self, n_f=32, ifactivate=False):
        super(CONV_OP, self).__init__()
        self.mylayers = []
        self.mylayers.append(tf.keras.layers.Conv3D(n_f, 3, strides=1, padding='same', use_bias=False))
        if ifactivate == True:
            self.mylayers.append(tf.keras.layers.ReLU())
        self.seq = tf.keras.Sequential(self.mylayers)

    def call(self, input):
        res = self.seq(input)
        return res


class exam_net(tf.keras.Model):
    def __init__(self):
        super(exam_net, self).__init__(name='exam_net')
        self.conv = CONV_OP(n_f=2, ifactivate=False)
        self.conv_t = CONV_OP(n_f=2, ifactivate=False)
        self.thres_coef = tf.Variable(tf.constant(-2, dtype=tf.float32), trainable=True, name='thres_coef')

    def call(self, x_in):
        [batch, Nt, Nx, Ny] = x_in.get_shape()

        x_in = tf.stack([tf.math.real(x_in), tf.math.imag(x_in)], axis=-1)
        x = self.conv(x_in)

        x_c = tf.complex(x[:, :, :, :, 0], x[:, :, :, :, 1])

        St, Ut, Vt = tf.linalg.svd(x_c, compute_uv=True, full_matrices=True)
        thres = tf.sigmoid(self.thres_coef) * St[..., 0]
        thres = tf.expand_dims(thres, -1)
        St = tf.nn.relu(St - thres)
        St = tf.linalg.diag(St)

        St = tf.dtypes.cast(St, tf.complex64)
        Vt_conj = tf.transpose(Vt, perm=[0, 1, 3, 2])
        Vt_conj = tf.math.conj(Vt_conj)
        US = tf.linalg.matmul(Ut, St)
        x_soft = tf.linalg.matmul(US, Vt_conj)
        
        print(np.isnan(St.numpy()).any())
        print(np.isnan(Ut.numpy()).any())
        print(np.isnan(Vt.numpy()).any())

        x_soft = tf.stack([tf.math.real(x_soft), tf.math.imag(x_soft)], axis=-1)
        x_out = self.conv_t(x_soft)

        output = x_out + x_in
        output = tf.complex(output[:, :, :, :, 0], output[:, :, :, :, 1])

        return output

 

if __name__ == '__main__':
    with tf.GradientTape() as g:
        # x_in  = tf.constant(np.load('1.npy'))
        x_in  = tf.ones([1, 5, 5, 5], dtype=tf.complex64)
        g.watch(x_in)

        net = exam_net()
        x_out = net(x_in)

        grad = g.gradient(x_out, [x_in, net.trainable_variables])
        print(grad[1][0])
```
```


### Relevant log output

_No response_</details>

**Comment by bhack** [comment]
> `tf2.8 reports error: Segmentation Fault (core dumped)` 

I cannot reproduce this on Colab GPU (TF 2.8)

Please note also that usually the fix in the older versions (<=1 year) will be done if that is a security bug or if the bug is critical and impacting larger community. In this case it is working fine or not reproducible with Tensorflow 2.8 you need to continue using this version.

**Comment by yhao-z** [comment]
* Yes, you're right. The standalone code can be running without any error as the input of the `exam_net` being all-one tensor. 
* When I use a large dataset to train the `exam_net`, __sometimes__ `Segmentation Fault` occurs.
* Maybe I should find __the specific tensor__ that cause the `Segmentation Fault` on `tf 2.8`, and issue it later.
* Thanks very much

**Comment by bhack** [comment]
> Maybe I should find the specific tensor that cause the Segmentation Fault on tf 2.8, and issue it later.

Yes if you can isolate this it would be ideal. Extending in a "test like" form it is the better solution:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/linalg/svd_op_test.py

Also in the case you can reproduce it only on your local env but not on Colab (e.g. it is specific of your env/hardware) it would be ideal to test it locally in a reproducible env:
https://www.tensorflow.org/install/docker

**Comment by yhao-z** [comment]
yes, thanks. I'll try it.

**Comment by bhack** [comment]
It could be nice if could extend the mentioned test contributing a PR to cover your case if you will find a way to reproduce it (on master/nightly).

**Comment by sushreebarsa** [comment]
@yhao-z I was not able to reproduce this issue on colab using TF v2.8.0,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/8e431256325b446520a309abbc5d6901/55840.ipynb).Could you please refer to this [comment](https://github.com/tensorflow/tensorflow/issues/55840#issuecomment-1117617224) above and and let us know the update on this?Thanks!
